### ISSUE TITLE
[REQUEST] Composite loss function

### ISSUE THREAD

**Issue by casadoj** [casadoj/neuralhydrology#9]
**Is your feature request related to a problem? Please describe.**
Create a custom loss function that focuses both in high and low flows.

**Describe the solution you'd like**

To run a true **MSE + Log-RMSE** hybrid loss, create a custom loss class in a Python script (e.g., `custom_losses.py` in your working directory) and register it with `neuralhydrology`:

```python
import torch
import torch.nn as nn
from neuralhydrology.training.losses import BaseLoss

class CompositeMSELogRMSE(BaseLoss):
    """Hybrid loss: Linear MSE (peaks) + Log-RMSE (recessions/baseflows)."""
    def __init__(self, eps: float = 0.001, alpha: float = 1.0):
        super().__init__()
        self.eps = eps
        self.alpha = alpha
        self.mse = nn.MSELoss()
        self.rmse = nn.MSELoss()

    def forward(self, y_hat: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
        # Linear MSE for high flows
        loss_linear = self.mse(y_hat, y)
        
        # Log-RMSE for low flows
        log_y_hat = torch.log(torch.clamp(y_hat, min=0) + self.eps)
        log_y = torch.log(torch.clamp(y, min=0) + self.eps)
        loss_log = torch.sqrt(self.rmse(log_y_hat, log_y) + 1e-8)
        
        return loss_linear + self.alpha * loss_log

```

#### How to use it in `neuralhydrology`:

1. Import and register your class in your entry-point execution script before calling `start_run()`:
```python
from neuralhydrology.training.losses import USER_LOSSES
from custom_losses import CompositeMSELogRMSE

# Register the loss name
USER_LOSSES["CompositeMSELogRMSE"] = CompositeMSELogRMSE

```

2. Set your YAML config file:
```yaml
loss: CompositeMSELogRMSE
```
